editor/editor/mode/visual.js

```diff
@@ -1,13 +1,22 @@
 /**
+ * External dependencies
+ */
+import { findIndex } from 'lodash';
+
+/**
  * Internal dependencies
  */
 import InserterButton from '../../inserter/button';
 
 function Blocks( { blocks, onChange } ) {
-	const onChangeBlock = ( index ) => ( changes ) => {
+	const onChangeBlock = ( uid ) => ( changes ) => {
+		const index = findIndex( blocks, { uid } );
 		const newBlock = {
```

Comment:
Mmm, [as documented](https://github.com/WordPress/gutenberg/tree/master/blocks#wpblocksregisterblock-slug-string-settings-object-), properties of `setAttributes` are applied as a partial patch, not a complete replacement. I like to think of it akin to React's `setState`. Otherwise it's very inconvenient to update a single attribute of a block in an input's event handler. I'd be okay with an additional `replaceAttributes` perhaps though if we need to support this case.